### PR TITLE
View specific recipe

### PR DESCRIPTION
**Pull request by Haoyuli2002**
I have implemented the method viewSpecificRecipe so that the user can view all the steps for a given recipe either by index or recipe title.

### CODE REVIEW

File: src/main/java/essenmakanan/command/AddRecipeCommand.java [hunk 1]
```diff
@@ -16,9 +16,22 @@ public AddRecipeCommand(String toAdd, RecipeList recipes) {
...
-        Recipe newRecipe = new Recipe(recipeTitle);
-        recipes.addRecipe(newRecipe);
-        ui.printAddRecipeSuccess(recipeTitle);
+        if (toAdd.contains("r/") & toAdd.contains("s/")) {
+            String[] allToAdd = toAdd.split("s/");
+            String recipeTitle = RecipeParser.parseRecipeTitle(allToAdd[0].trim());
+            String[] steps = new String[allToAdd.length-1];
+            for (int i = 1; i < allToAdd.length; i++) {
+                steps[i-1] = allToAdd[i].trim();
+            }
```
Comment (charkty) [inline src/main/java/essenmakanan/command/AddRecipeCommand.java:25]: can start from i=0 for the for loop, then just need to use steps[i]

File: src/main/java/essenmakanan/command/AddRecipeCommand.java [hunk 2]
```diff
@@ -16,9 +16,22 @@ public AddRecipeCommand(String toAdd, RecipeList recipes) {
 
     @Override
     public void executeCommand() {
-        String recipeTitle = RecipeParser.parseRecipeTitle(toAdd);
-        Recipe newRecipe = new Recipe(recipeTitle);
-        recipes.addRecipe(newRecipe);
-        ui.printAddRecipeSuccess(recipeTitle);
+        if (toAdd.contains("r/") & toAdd.contains("s/")) {
+            String[] allToAdd = toAdd.split("s/");
+            String recipeTitle = RecipeParser.parseRecipeTitle(allToAdd[0].trim());
```
Comment (charkty) [inline src/main/java/essenmakanan/command/AddRecipeCommand.java:21]: toAdd shouldnt have "r/", if it does, should be removed i parser class. Hence, dont need Recipe.Parser.parseRecipeTitle(...)

File: src/main/java/essenmakanan/command/AddRecipeCommand.java [hunk 3]
```diff
@@ -16,9 +16,22 @@ public AddRecipeCommand(String toAdd, RecipeList recipes) {
...
+            String[] steps = new String[allToAdd.length-1];
+            for (int i = 1; i < allToAdd.length; i++) {
+                steps[i-1] = allToAdd[i].trim();
+            }
+            Recipe newRecipe = new Recipe(recipeTitle, steps);
+            recipes.addRecipe(newRecipe);
+            ui.printAddRecipeSuccess(recipeTitle);
+        } else {
+            String recipeTitle = RecipeParser.parseRecipeTitle(toAdd);
```
Comment (charkty) [inline src/main/java/essenmakanan/command/AddRecipeCommand.java:30]: same as above comments^

File: src/main/java/essenmakanan/recipe/RecipeList.java [hunk 4]
```diff
@@ -32,4 +35,38 @@ public Recipe getRecipe(int index) {
         return recipes.get(index);
     }
 
+    public void viewSpecificRecipe(int index) {
+        Ui.drawDivider();
+        if (index < 0 | index >= recipes.size()) {
```
Comment (charkty) [inline src/main/java/essenmakanan/recipe/RecipeList.java:40]: i think the or symbol is "||", one line is for bitwise

File: src/main/java/essenmakanan/recipe/RecipeList.java [hunk 5]
```diff
@@ -32,4 +35,38 @@ public Recipe getRecipe(int index) {
...
+            System.out.println("We have " + recipes.size() + "recipes right now and the given input is invalid.");
+            return;
+        }
+        Recipe recipe = recipes.get(index-1);
+        listAllSteps(recipe);
+    }
+
+    public void viewSpecificRecipe(String title) {
+        Ui.drawDivider();
+        Recipe recipe = recipes.stream().filter(recipe1 -> recipe1.getTitle().equals(title)).findFirst().orElse(null);
```
Comment (charkty) [inline src/main/java/essenmakanan/recipe/RecipeList.java:50]: can add a new line for each function called. 
image

and can change recipe1 name to filteredRecipe

**Review by charkty — COMMENTED** [review]
Changes to be made.

File: src/main/java/essenmakanan/command/AddRecipeCommand.java (hunk 2 above)
Comment (Haoyuli2002) [reply src/main/java/essenmakanan/command/AddRecipeCommand.java:21]: Actually, toAdd contains "r/", you may refer this screenshot
Screenshot 2023-10-26 at 15 48 34

File: src/main/java/essenmakanan/recipe/RecipeList.java (hunk 4 above)
Comment (Haoyuli2002) [reply src/main/java/essenmakanan/recipe/RecipeList.java:40]: modified! Thanks for the notice

File: src/main/java/essenmakanan/recipe/RecipeList.java (hunk 5 above)
Comment (Haoyuli2002) [reply src/main/java/essenmakanan/recipe/RecipeList.java:50]: Modified, it's more readable now :D

**Pull request closed without merging**